controller/eks-cluster-config-handler.go

```diff
@@ -495,6 +455,56 @@ func (h *Handler) create(config *v13.EKSClusterConfig, sess *session.Session, ek
 	return h.eksCC.UpdateStatus(config)
 }
 
+func (h *Handler) generateAndSetNetworking(svc *cloudformation.CloudFormation, config *v13.EKSClusterConfig) (*v13.EKSClusterConfig, error) {
+	if len(config.Status.Subnets) != 0 {
+		// networking fields have already been set
+		return config, nil
+	}
+
+	if len(config.Spec.Subnets) != 0 {
+		logrus.Infof("VPC info provided, skipping vpc/subnet/securitygroup creation")
+		config = config.DeepCopy()
+		// copy networking fields to status
+		config.Status.Subnets = config.Spec.Subnets
+		config.Status.SecurityGroups = config.Spec.SecurityGroups
+		config.Status.NetworkFieldsSource = "provided"
+		var err error
+		config, err = h.eksCC.UpdateStatus(config)
+		if err != nil {
+			return config, err
+		}
+	} else {
```

Comment:
Can we remove the else statement here? It seems like at the moment we plan to `return h.eksCC.UpdateStatus(config)` regardless of the output (either returning `config, err` where `err != nil` in line 474 or `config, nil` where `err == nil` in line 505).